tests/legacy/system/test_edb_crosstalk_scanner.py

```diff
@@ -0,0 +1,72 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Tests related to Edb
+"""
+
+import os
+
+import pytest
+
+pytestmark = [pytest.mark.system, pytest.mark.legacy]
+
+
+class TestClass:
+    @pytest.fixture(autouse=True)
+    def init(self, legacy_edb_app, local_scratch, target_path, target_path2, target_path4):
+        self.edbapp = legacy_edb_app
+        self.local_scratch = local_scratch
+        self.target_path = target_path
+        self.target_path2 = target_path2
+        self.target_path4 = target_path4
+
+    def test_create_impedance_scan(self):
+        xtalk_scan = self.edbapp.siwave.create_crosstalk_config_file(scan_type="impedance")
+        for net in list(self.edbapp.nets.signal.keys()):
+            xtalk_scan.impedance_scan.add_single_ended_net(
+                name=net, nominal_impedance=45.0, warning_threshold=40.0, violation_threshold=30.0
+            )
+        xtalk_scan.file_path = os.path.join(self.local_scratch.path, "test_impedance_scan.xml")
+        assert xtalk_scan.write_xml()
```

Comment:
It only checks if the xml file exists. Can you also check if content of the file is valid?

Reply:
done